### ISSUE TITLE
Guest Executable Status

### ISSUE THREAD

**Issue by aliazamrana** [Azure/service-fabric-issues#889]
I have a senstive application that needs monitoring of the guest executables deployed on the Service Fabric. Now I have two questions one is that is it possible to have a guest executable that fails to deploy? And if yes how can I find out and is it possible to find out through QueryManager in FabricClient.
I can find out that my applications are deployed on which node and their status is returned fine but How can I find out about the guest executables that are not deployed.

Thanks

**Comment by rishirsinha** [comment]
@aliazamrana 

Can you explain what you mean by fails to deploy? Are saying that we provision it and some how create service fabric application / service command fails? 

It would also bee helpful if you can explain what you mean by sensitive application.

**Comment by aliazamrana** [comment]
Let me give a complete scenario that. I have a service fabric project that has a few guest executables on it and my stateful service provides an api for them and then those guest executables are used to do trading depending upon the requests they receive from my stateful service. 
Now as you see that these guest executables are used for trading it is very important to check their status. Now I have properly obtained the status of the nodes and I am able to find out on which nodes my guest executables are deployed. 
Now I wanted to know that is it possible that these guest executables may not get deployed due to some error but my stateful service is running fine amd is properly deployed. How can I find out which guest executables are not deployed currently I am getting all the required information about the deployed applications through the QueryManager in FabricClient for C#.
I hope I was clear this time. 
PS sorry for inconvenience

**Comment by rishirsinha** [comment]
Are you deploying the services all as a single application? Or is the stateful service creating the guest executables on demand?

The deployment can always fail. But you will be able to see errors. Any time create application fails, it will say so. if the create application has succeeded and instances go down, we will restart them and give a health warning. Either way you will know what is happening.

**Comment by aliazamrana** [comment]
Yes all the services are deployed in a single application and it deploys fine but I wanted to know how to verify if some service in the application fails to deploy how can I query that through query manager in fabric client in c#

**Comment by rishirsinha** [comment]
In that case I am assuming you have instance count for the services specified. If that is the case you can use the following:

https://docs.microsoft.com/en-us/dotnet/api/system.fabric.fabricclient.queryclient.getdeployedreplicalistasync?view=azure-dotnet#System_Fabric_FabricClient_QueryClient_GetDeployedReplicaListAsync_System_String_System_Uri_

to find out how many instances are available and make sure that it matches the expected count.

**Comment by aliazamrana** [comment]
Yes was doing it the same way my process involved getting deployed list and getting the list of services and then I matched those together to see if one was not in the other. I was hoping there was some other way to just query or filter out the service which failed to deploy

This is how I got the deployed services

 `var deployedServices = fabricClient.QueryManager.GetDeployedServicePackageListAsync(node.NodeName, application.ApplicationName).Result;`

Next did this to get the services and see how total services 

`var services = fabricClient.QueryManager.GetServiceListAsync(app.ApplicationName).Result;`

Next I matched both lists of 

> services 

and 

> deployedServices 

Is this the right way?
